spark_api.gemspec

```diff
@@ -31,30 +31,16 @@ Gem::Specification.new do |s|
   s.default_executable = %q{spark_api}
   s.require_paths      = ["lib"]
   
-  s.add_dependency 'faraday', '~> 0.9.0'
-  s.add_dependency 'multi_json', '~> 1.0'
-  s.add_dependency 'json', '~> 1.7'
-  s.add_dependency 'builder', '>= 2.1.2', '< 4.0.0'
-  s.add_dependency 'will_paginate', '>= 3.0.pre2', '< 4.0.0'
-  s.add_dependency 'highline', '>= 1.0'
-
-  # spark_api doesn't use public_suffix directly. spark_api uses Webmock, which 
-  # uses addressable, which uses public_suffix. Bundler has suddenly started 
-  # trying to install public_suffix 2.0.4, which requires Ruby 2. When spark_api 
-  # starts to require Ruby 2, this dependency can be removed.
-  s.add_dependency 'public_suffix', '~> 1.4.6'
+  s.add_dependency 'faraday'
+  s.add_dependency 'multi_json'
+  s.add_dependency 'json'
+  s.add_dependency 'will_paginate'
 
   # TEST GEMS
-  s.add_development_dependency 'rake', '~> 0.9.2'  
-  s.add_development_dependency 'rspec', '~> 2.14.0'
-  s.add_development_dependency 'webmock', '~> 1.9'
-  s.add_development_dependency 'typhoeus', '~> 0.3'
-  s.add_development_dependency 'ci_reporter', '~> 1.7.0'
-  s.add_development_dependency 'rb-readline'
-  s.add_development_dependency 'rb-fsevent'
-  s.add_development_dependency 'simplecov'
+  s.add_development_dependency 'rake'
```

Comment:
I think I'm OK with keeping the dependencies pretty wide open, but you may want to be tighter on dev dependencies. This is of course assuming we don't checking a lock file.

I don't have a clear solution here, but there is a definite problem with working on a gem because one's local versions are vastly different from the main author.

Reply:
Perhaps a guideline here is to limit to >= an exact version that is known to work.

Reply:
If I understand you correctly, you're talking about the problem that happens when your local copy has preinstalled gems, and the tests all pass, but the from-scratch job on Travis or our build server fails, either when installing deps or because a changed dependency caused a test to fail?

Reply:
Yup, exactly. Everything works fine up until a new version of the dependency is released, and trashes your next build. It's a nuisance, and can be downright maddening to figure out the last good list of deps.